### PR TITLE
Prevent currently logged user deletion

### PR DESCRIPTION
**Pull request by Zales0123**

### CODE REVIEW

**Comment by pjedrzejewski** [comment]
Thank you Mateusz! :+1:

File: src/Sylius/Bundle/CoreBundle/EventListener/UserDeleteListener.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        $user = $event->getSubject();
+
+        if (!$user instanceof UserInterface) {
+            throw new UnexpectedTypeException(
+                $user,
+                'FOS\UserBundle\Model\UserInterface'
+            );
+        }
+
+        if ($this->securityContext->getToken()->getUsername() === $user->getUsernameCanonical()) {
```
Comment (stloyd) [inline src/Sylius/Bundle/CoreBundle/EventListener/UserDeleteListener.php]: This is wrong, username is something that can be changed, you __must__ check identifier not username!

Comment (pjedrzejewski) [inline src/Sylius/Bundle/CoreBundle/EventListener/UserDeleteListener.php]: @Zales0123 Yeah, let's check the id, could you update this in a separate PR, thanks!

File: src/Sylius/Bundle/CoreBundle/EventListener/UserDeleteListener.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    protected $securityContext;
+    protected $session;
+    protected $router;
+    protected $redirectTo;
+
+    public function __construct(SecurityContext $securityContext, UrlGeneratorInterface $router, SessionInterface $session, $redirectTo)
+    {
+        $this->securityContext = $securityContext;
+        $this->session = $session;
+        $this->router = $router;
```
Comment (stloyd) [inline src/Sylius/Bundle/CoreBundle/EventListener/UserDeleteListener.php]: Both router & redirect options should be remove as those variables are never used...

**Pull request merged**